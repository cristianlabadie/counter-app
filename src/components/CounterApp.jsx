import PropTypes from "prop-types";
import { useState } from "react";

export const CounterApp = ({ value }) => {
  let [counter, setCounter] = useState(value);
  const handleIncrement = () => setCounter(counter + 1);
  const handleDecrement = () => setCounter(counter - 1);
  const handleReset = () => setCounter(value);

  return (
    <>
      <h1>CounterApp</h1>
      <h2 data-testid="test-counter">{counter}</h2>

      <button onClick={handleIncrement}>+</button>
      <button onClick={handleDecrement}>-</button>
      <button aria-label="btn-reset" onClick={handleReset}>
        Reset
      </button>
    </>
  );
};

CounterApp.propTypes = {
  value: PropTypes.number.isRequired,
};
