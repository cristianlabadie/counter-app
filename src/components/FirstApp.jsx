import PropTypes from "prop-types";

export const FirstApp = ({ title, subtitle, name }) => {
  return (
    <>
      <h2 data-testid="test-title"> {title} </h2>
      <p data-testid="test-subtitle">{subtitle}</p>
      <p>{name}</p>
    </>
  );
};

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string,
  name: PropTypes.string,
};

FirstApp.defaultProps = {
  subtitle: "Soy un subtitulo",
};
