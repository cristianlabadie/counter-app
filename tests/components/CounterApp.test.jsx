import { fireEvent, render, screen } from "@testing-library/react";
import { CounterApp } from "../../src/components/CounterApp";

describe("Pruebas en <CounterApp />", () => {
  const value = 100;

  test("debe de hacer match con el snapshot", () => {
    const { container } = render(<CounterApp value={value} />);
    expect(container.innerHTML).toMatchSnapshot();
  });

  test("debe de mostrar el valor inicial de 100 <CounterApp value={100} />", () => {
    render(<CounterApp value={value} />);
    expect(parseInt(screen.getByTestId("test-counter").innerHTML)).toBe(value);
  });

  test("debe de incrementar con botón +", () => {
    render(<CounterApp value={value} />);
    fireEvent.click(screen.getByText("+"));
    expect(screen.getByText(value + 1)).toBeTruthy();
  });

  test("debe de decrementar con botón -", () => {
    render(<CounterApp value={value} />);
    fireEvent.click(screen.getByText("-"));
    expect(screen.getByText(value - 1)).toBeTruthy();
  });

  test("debe de resetear con botón reset después de haber incrementado en 1", () => {
    render(<CounterApp value={value} />);
    fireEvent.click(screen.getByText("+"));
    expect(screen.getByText(value + 1)).toBeTruthy();

    fireEvent.click(screen.getByRole("button", { name: "btn-reset" }));
    expect(screen.getByText(value).innerHTML).toBe("100");
  });
});
