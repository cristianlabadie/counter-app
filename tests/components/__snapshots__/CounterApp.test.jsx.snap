// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <CounterApp /> debe de hacer match con el snapshot 1`] = `"<h1>CounterApp</h1><h2 data-testid="test-counter">100</h2><button>+</button><button>-</button><button aria-label="btn-reset">Reset</button>"`;
